src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/SimpleTypeModelBinder.cs

```diff
@@ -47,23 +47,30 @@ public Task BindModelAsync(ModelBindingContext bindingContext)
             {
                 var value = valueProviderResult.FirstValue;
 
-                object model = null;
-                if (!string.IsNullOrWhiteSpace(value))
-                {
-                    model = _typeConverter.ConvertFrom(
-                        context: null,
-                        culture: valueProviderResult.Culture,
-                        value: value);
-                }
-
+                object model;
                 if (bindingContext.ModelType == typeof(string))
                 {
-                    var modelAsString = model as string;
-                    if (bindingContext.ModelMetadata.ConvertEmptyStringToNull &&
-                        string.IsNullOrEmpty(modelAsString))
+                    // Already have a string. No further conversion required but handle ConvertEmptyStringToNull.
+                    if (bindingContext.ModelMetadata.ConvertEmptyStringToNull && string.IsNullOrWhiteSpace(value))
```

Comment:
MVC 5 and ASP.NET 3 both use `string.IsNullOrWhiteSpace()` in their analogous code. Same for `ValueProviderResultExtensions.ConvertTo()` which this code used prior to 4e97c72. (It's that commit that broke `!ConvertEmptyStringToNull` and added useless type conversions from `string` to `string`.)